test/unit/org/apache/cassandra/index/sai/plan/EstimatedRowCountTest.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.plan;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import org.apache.cassandra.Util;
+import org.apache.cassandra.cql3.Operator;
+import org.apache.cassandra.db.ColumnFamilyStore;
+import org.apache.cassandra.db.ReadCommand;
+import org.apache.cassandra.index.sai.QueryContext;
+import org.apache.cassandra.index.sai.SAITester;
+import org.apache.cassandra.index.sai.SAIUtil;
+import org.apache.cassandra.index.sai.disk.format.Version;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+
+public class EstimatedRowCountTest extends SAITester
+{
+    private int queryOptLevel;
+
+    @Before
+    public void setup()
+    {
+        queryOptLevel = QueryController.QUERY_OPT_LEVEL;
+        QueryController.QUERY_OPT_LEVEL = 0;
+    }
+
+    @After
+    public void teardown()
+    {
+        QueryController.QUERY_OPT_LEVEL = queryOptLevel;
+    }
+
+    private Object getFilterValue(String type, int value)
```

Comment:
nit: value is always 25; consider removing the param

Reply:
I've seen lint's warning about it. IMHO it's more readable to pass the value and keep the parameters.
I will see if I add another test.

Reply:
I added more tests where it varies.